go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -1005,10 +1005,10 @@ func (bnc *BaseNetworkController) DeleteResourceCommon(objType reflect.Type, obj
 			return fmt.Errorf("could not cast obj of type %T to *knet.NetworkPolicy", obj)
 		}
 		netinfo, err := bnc.networkManager.GetActiveNetworkForNamespace(knp.Namespace)
-		if err != nil {
+		if err != nil && !errors.Is(err, util.NewInvalidPrimaryNetworkError(knp.Namespace)) {
```

Comment:
NIT: Can you please add a comment explaining why we accept this error here?

Reply:
done.